src/components/structures/MatrixChat.js

```diff
@@ -1042,7 +1071,9 @@ module.exports = React.createClass({
                     registrationUrl={this.props.registrationUrl}
                     onLoggedIn={this.onRegistered}
                     onLoginClick={this.onLoginClick}
-                    onRegisterClick={this.onRegisterClick} />
+                    onRegisterClick={this.onRegisterClick}
+                    onCancelClick={ MatrixClientPeg.get() && MatrixClientPeg.get().isGuest() ? this.onReturnToGuestClick : null }
```

Comment:
How does this work? Will this not hide the back-to-app div if the user does anything to cause the matrix client to be replaced, like changing home servers?